### PR TITLE
replacing eval()s to avoid 'eval injection' and improve HHVM Repo Author...

### PR DESCRIPTION
**Pull request by colin-kiegel**
...itative compatibility

### CODE REVIEW

**Comment by colin-kiegel** [comment]
again I performed extra tests, which can be found here: https://gist.github.com/colin-kiegel/b8e1ba2c1a1c414c4379

File: Modules/SurveyQuestionPool/phplot/phplot.php [hunk 1]
```diff
@@ -669,7 +669,7 @@ function SetDefaultDashedStyle($which_style)
         }
 
         // Build the string to be eval()uated later by SetDashedStyle()
-        $this->default_dashed_style = 'array( ';
+        $this->default_dashed_style = '[ '; // switching from PHP-Array to JSON-Notation
```
Comment (chfsx) [inline Modules/SurveyQuestionPool/phplot/phplot.php]: Since we still support PHP 5.3 this notation can''t be used

**Comment by chfsx** [comment]
Hi Colin. PHPlot is an external library, I'm not sure whether we should make changes here...

**Comment by Amstutz** [comment]
I agree with Fabian. Since since the development on Phplot seems still active, you should make the request there. To keep our dependencies clean (patch free) we should always try to play our fixes in external libraries back to the repo of the library itself. I think your changes have good chances to be accepted there.

Since the version of Phplot is of 2008-01-13, we should check to update to a current version anyways, which might already solve the problem. 

@chfsx: We should think about a proper way of including external libraries as submodule (or similar) to have a cleaner build an proper relations with our dependencies in future.

**Comment by colin-kiegel** [comment]
ok, I agree - phplot seems actively maintained. I will proposed it upstream (https://sourceforge.net/p/phplot/bugs/170/). However it might take 'a while' until this will (a) be accepted there (b) released officially and (c) integrated in ILIAS.

The other two libraries PEAR and math eval don't seem to be actively maintained. I think it would be ok to include these really small changes left.

**Comment by alex40724** [comment]
If patches of third party libs are necessary, this must be documented. See e.g. Services/PEAR/lib/ReleaseInfo.php

**Comment by colin-kiegel** [comment]
done - class.MathEval.php didn't have a ReleaseInfo, so I created one.

**Comment by alex40724** [comment]
Björn, since you rely on MathEval I leave it to you to merge this one if you accept it. For me the PEAR patch is ok.

**Comment by chfsx** [comment]
+1 for merging if possible for you @bheyser

**Pull request merged**